lib/Parse/ParseDecl.cpp

```diff
@@ -6623,6 +6626,8 @@ Parser::parseDeclSubscript(SourceLoc StaticLoc,
                    Flags, StaticLoc, Attributes,
                    ElementTy.get(), Indices.get(), Decls);
 
+  Subscript->setEndLoc(getEndOfPreviousLoc().getAdvancedLoc(-1));
```

Comment:
I'm not thrilled with setting the end loc here, because if we ever add more early returns it's easy to forget it needs this as well.  Is there something we can do to ensure it's always set?

Reply:
I like that worry. I'll think on it.

Reply:
OK, I'm trying to changes (not pushed yet):
1. Don't need to set the end loc before the first return, which does not return the SubscriptDecl.
2. Set the end loc as soon as the parsing is finished, moving that line up. And adding a comment.

I'll see how this works.